classes/client/s3_client.php

```diff
@@ -331,6 +355,33 @@ public function define_client_section($mform, $config) {
 
         $mform->addElement('select', 's3_region', get_string('settings:aws:region', 'tool_objectfs'), $regionoptions);
         $mform->addHelpButton('s3_region', 'settings:aws:region', 'tool_objectfs');
+
+        $mform = $this->define_proxy_config($mform, $config);
+
+        return $mform;
+    }
+
+    private function define_proxy_config($mform, $config) {
```

Comment:
Like this abstraction :+1:

Reply:
Isn't the point of this to honor the pre existing moodle proxy config which already has it's own config here:

https://github.com/moodle/moodle/blob/master/admin/settings/server.php#L102-L112

Generally speaking if we use the built in moodle curl library you get this for free:

https://github.com/moodle/moodle/blob/master/lib/filelib.php#L2946

But I suspect this will not be the case because we use GuzzleHttp but it would be case by case depending on S3 vs azure etc?

Either way this whole patch will need a rebase / rework against the dev branch which has moved on a fair bit